src/codeu/chat/client/android/app/src/main/java/com/google/codeu/chatme/presenter/ChatActivityPresenter.java

```diff
@@ -0,0 +1,76 @@
+package com.google.codeu.chatme.presenter;
+
+import android.util.Log;
+
+import com.google.codeu.chatme.model.Conversation;
+import com.google.codeu.chatme.utility.FirebaseUtil;
+import com.google.codeu.chatme.view.adapter.ChatListAdapter;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.Query;
+import com.google.firebase.database.ValueEventListener;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+/**
+ * Following MVP design pattern, this class encapsulates the functionality to
+ * store and retrieve data related to current user's conversations from Firebase
+ * database
+ */
+public class ChatActivityPresenter implements ChatActivityInteractor {
+
+    private static final String TAG = ChatActivityPresenter.class.getName();
+
+    private DatabaseReference mRootRef = FirebaseDatabase.getInstance().getReference();
+
+    /**
+     * {@link ChatListAdapter} reference to update list of conversations
+     */
+    private final ChatListAdapter view;
+
+    /**
+     * Constructor to accept a reference to a recycler view adapter to bind
+     * conversation data to views
+     *
+     * @param view {@link ChatListAdapter} reference
+     */
+    public ChatActivityPresenter(ChatListAdapter view) {
+        this.view = view;
+    }
+
+    /**
+     * Loads conversations of the current user from Firebase
+     */
+    public void loadConversations() {
+        Query conversationsQuery = mRootRef.child("conversations").orderByChild("timeCreated");
+        conversationsQuery.addValueEventListener(new ValueEventListener() {
+
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                ArrayList<Conversation> conversations = new ArrayList<>();
+
+                for (DataSnapshot data : dataSnapshot.getChildren()) {
+                    Conversation conv = data.getValue(Conversation.class);
+                    if (conv.getParticipants().contains(FirebaseUtil.getCurrentUserUid())) {
```

Comment:
Fetching all conversations and using client-side filtering is a likely performance problem down the road (and will inhibit other optimizations like fetching only a limited number of recent conversations.)  If it needs to be this way, I'd like to see some documentation of what lead to this design decision.  I think it would be worth revisiting the database schema to try and support a filtered query here.